HW05-TestAnnotation/src/main/java/ru/otus/TestAnnotation/TestRunner.java

```diff
@@ -0,0 +1,56 @@
+package ru.otus.TestAnnotation;
+
+import ru.otus.TestAnnotation.Annotation.After;
+import ru.otus.TestAnnotation.Annotation.Before;
+import ru.otus.TestAnnotation.Annotation.Test;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestRunner {
+    public static <T> void run(Class<T> clazz) {
+        List<Method> methodsTest = getMethods(clazz, Test.class);
+        List<Method> methodsAfter = getMethods(clazz, After.class);
+        List<Method> methodsBefore = getMethods(clazz, Before.class);
+
+        T testObj = null;
+
+        int passed = 0;
+        for (Method m : methodsTest) {
+            try {
+                testObj = clazz.getDeclaredConstructor().newInstance();
+                for (Method methodBefore : methodsBefore) {
+                    methodBefore.invoke(testObj);
+                }
+
+                m.invoke(testObj);
+                passed += 1;
+            } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
```

Comment:
```suggestion
            } catch (Exception e) {
```

Здесь считаем что тест упал при любом исключении